server/container/util/src/test/java/org/apache/james/util/concurrency/ConcurrentTestRunnerTest.java

```diff
@@ -132,4 +133,79 @@ public void consume(int threadNumber, int step) throws Exception {
         assertThat(concurrentTestRunner.awaitTermination(DEFAULT_AWAIT_TIME, TimeUnit.MILLISECONDS)).isTrue();
         assertThat(queue).containsOnly("0:0", "0:1", "1:0", "1:1");
     }
+
+    @Test
+    public void runShouldNotThrowOnExceptions() throws Exception {
+        int operationCount = 2;
+        int threadCount = 2;
+
+        ConcurrentTestRunner concurrentTestRunner = new ConcurrentTestRunner(threadCount, operationCount,
+            new ConcurrentTestRunner.BiConsumer() {
+                @Override
+                public void consume(int threadNumber, int step) throws Exception {
+                    throw new RuntimeException();
+                }
+            })
+            .run();
+
+        assertThat(concurrentTestRunner.awaitTermination(DEFAULT_AWAIT_TIME, TimeUnit.MILLISECONDS)).isTrue();
+    }
+
+    @Test
+    public void noExceptionsShouldNotThrowWhenNoExceptionGenerated() throws Exception {
+        int operationCount = 2;
+        int threadCount = 2;
+
+        ConcurrentTestRunner concurrentTestRunner = new ConcurrentTestRunner(threadCount, operationCount,
+            new ConcurrentTestRunner.BiConsumer() {
+                @Override
+                public void consume(int threadNumber, int step) throws Exception {
+
+                }
+            })
+            .run();
+
+        concurrentTestRunner.awaitTermination(DEFAULT_AWAIT_TIME, TimeUnit.MILLISECONDS);
+
+        concurrentTestRunner.noExceptions();
+    }
+
+    @Test
+    public void runShouldThrowOnExceptions() throws Exception {
```

Comment:
I think I get it don't you mean noExceptionShouldThrowOnException instead of runShouldThrowOnExceptions